maven.grammar/test/unit/src/org/netbeans/modules/maven/hyperlinks/HyperlinkProviderImplTestForMultiPomProjects.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.modules.maven.hyperlinks;
+
+import java.io.File;
+import java.io.IOException;
+import javax.swing.text.BadLocationException;
+import javax.swing.text.Document;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.netbeans.api.lexer.Language;
+import org.netbeans.api.xml.lexer.XMLTokenId;
+import org.netbeans.junit.NbTestCase;
+import org.netbeans.lib.editor.hyperlink.spi.HyperlinkType;
+import org.netbeans.modules.maven.grammar.POMDataObject;
+import org.openide.cookies.EditorCookie;
+import org.openide.filesystems.FileObject;
+import org.openide.filesystems.LocalFileSystem;
+import org.openide.loaders.DataObject;
+
+/**
+ * Additional Unit tests for the HyperlinkProvider for pom.xml files when used in a 
+ * hierarchy of maven projects.
+ */
+public class HyperlinkProviderImplTestForMultiPomProjects extends NbTestCase {
+    
+    private static final int START_DEPENDENCY_GROUP_ID_VALUE = 538;
```

Comment:
A suggestion: instead of hardcoding indexes, it may be helpful to place some special marks into the test data, which will be preprocessed out by the test. In your case, it could help to place "{token}" and then build an offset Map<String, Integer> while processing the {.*} out of the test data template.

Gets more important with each test data change.

Reply:
I've looked at, and I believe resolved all the items you mentioned except for this one.  I understand where your coming from on this one (working out the indexes for both tests for the existing functionality and the new one was a bit of effort). 

I'm just concerned about polluting the tests with a logic that reads in the file first, references each "placeholder" and removes it before the tests start, only to fix the files upon test completion... (unless I'm missing something here...)